dns.tf

```diff
@@ -0,0 +1,29 @@
+resource "digitalocean_domain" "top_level_domains" {
+    for_each = toset(var.top_level_domains)
+    name = each.value
+}
+
+resource "digitalocean_record" "a_records" {
+  for_each = toset(var.top_level_domains)
+  domain = each.value
+  type   = "A"
+  ttl = 60
+  name   = "@"
+  value  = digitalocean_loadbalancer.ingress_load_balancer.ip
+  depends_on = [
+    digitalocean_domain.top_level_domains,
+    kubernetes_ingress.default_cluster_ingress
```

Comment:
This line causes the following error:

```console
[2022-09-09T12:01:18.313Z] Terraform has been successfully initialized!
[2022-09-09T12:01:19.254Z] ╷
[2022-09-09T12:01:19.254Z] │ Error: Reference to undeclared resource
[2022-09-09T12:01:19.254Z] │ 
[2022-09-09T12:01:19.254Z] │   on dns.tf line 15, in resource "digitalocean_record" "a_records":
[2022-09-09T12:01:19.254Z] │   15:     kubernetes_ingress.default_cluster_ingress
[2022-09-09T12:01:19.254Z] │ 
[2022-09-09T12:01:19.254Z] │ A managed resource "kubernetes_ingress" "default_cluster_ingress" has not
[2022-09-09T12:01:19.254Z] │ been declared in the root module.
```

Not sure why do you want to add this dependency as we do not declare any Terraform resource of type `kubernetes_ingress` in this project (and not sure why we would want)?

Reply:
It was for a following part of https://ryderdamen.medium.com/deploying-kubernetes-web-servers-to-digital-ocean-with-tls-and-terraform-2ccba95c5a3c, my bad